core/Updates/5.0.0-b1.php

```diff
@@ -136,4 +145,12 @@ private function hasNewIndex(): bool
     {
         return DbHelper::tableHasIndex($this->tableName, $this->newIndexName);
     }
+
+    private function isTableInnoDb(): bool
+    {
+        $sql = "SHOW TABLE STATUS WHERE NAME='{$this->tableName}'";
+        $result = Db::fetchRow($sql);
+
+        return strtolower($result['Engine']) === 'innodb';
```

Comment:
Engine can be null in some cases such as when the table is damaged and needs repairing, so this would prevent incorrect type deprecation.

```suggestion
        return strtolower($result['Engine'] ?? '') === 'innodb';
```